code/modules/reagents/chemistry/reagents/toxin_reagents.dm

```diff
@@ -11,6 +11,15 @@
 	var/toxpwr = 1.5
 	var/silent_toxin = FALSE //won't produce a pain message when processed by liver/life() if there isn't another non-silent toxin present.
 
+/**
+  * Should a human's liver, in normal conditions, filter this toxin?
+  *
+  * Defaults to TRUE by default, may be overriden by specific toxins
+  * that can be beneficial under certain conditions.
+  */
+/datum/reagent/toxin/proc/liver_filter(mob/living/carbon/C, obj/item/organ/liver/L)
+	. = TRUE
```

Comment:
just return what the heck is this

Reply:
why not put this on base reagent, auto return false, and have toxins auto return true?

Reply:
The `.` return is a pattern I remember Remie encouraging, because alledgedly, "innate" returns were less expensive than "called" returns (using the `return` keyword).

And it's not on the base reagent because the liver filters by the /datum/reagent/toxin path. I could change the liver to filter all reagents, but it makes a number of assumptions of toxins (like silent vs noisy toxins).

Reply:
i would need to see profiler differences, that seems like deep-level microopt.

You would ideally replace the typecheck with just the liver check :)